runtime/src/bank.rs

```diff
@@ -1110,6 +1115,45 @@ impl Bank {
         )
     }
 
+    fn collect_rent(
+        &self,
+        txs: &[Transaction],
+        iteration_order: Option<&[usize]>,
+        loaded_accounts: &mut [Result<TransactionLoadResult>],
+    ) {
+        let mut collected_rent = 0;
+        let mut seen_credit_only_account: HashSet<Pubkey> = HashSet::new();
+        for (tx, loaded_account) in
+            OrderedIterator::new(txs, iteration_order).zip(loaded_accounts.iter())
+        {
+            if loaded_account.is_err() {
+                continue;
+            }
+            let (_account, _loaders, _credits, rents) = loaded_account.as_ref().unwrap();
+
+            let message = tx.message();
+
+            for ((i, key), rent) in message
+                .account_keys
+                .iter()
+                .enumerate()
+                .filter(|(_i, key)| !message.program_ids().contains(&key))
+                .zip(rents.iter())
+            {
+                if seen_credit_only_account.contains(&key) {
+                    continue;
+                }
+                collected_rent += rent;
```

Comment:
this won't work.  we can't actually debit the CO accounts during the TX, we have to do this when we reconcile CO credits

Reply:
why? `verify_instruction` func runs per instruction, and it runs before this func. So, this shouldn't be an issue.

OR 

Am I missing something?

Reply:
the thing you're missing is that we don't actually store() the CO accounts, we hang onto the credits until the end of the slot.

rent for CO accounts has to be done in a separate pass at the end of the slot